### PR TITLE
colw/2446 validate delegation total

### LINKED ISSUE #2446: Don't let user sign tx if not enough funds
1. User Account has 0.02 atoms
2. user tries to delegate 0.01 atom. Network fee is 0.02. Total amount in tx 0.03
3. Click "Next"

There could be an easy client-side check that I don't have enough funds.
Instead, the user is asked for a password and the app shows "Incorrect password error" after submit.
Very misleading.

### PR DESCRIPTION
**Pull request by colw**
Closes #2446 

**Description:**

Action Modal should now validate the total amount invoiced, including the network fees.

There will be an error displayed and the 'Next' button will be disabled if the value is incorrect.

Screenshot 2019-05-06 at 19 40 53

Thank you! 🚀

---

For contributor:

- [x] Added changes entries. Run `yarn changelog` for a guided process.
- [x] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/components/common/ActionModal.vue [hunk 1]
```diff
@@ -246,6 +254,11 @@ export default {
     balance() {
       return this.liquidAtoms
     },
+    totalInvoice() {
+      return (
+        Number(this.amount) + Number(this.gasPrice) * Number(this.gasEstimate)
```
Comment (colw) [inline src/components/common/ActionModal.vue]: @fedekunze @faboweb Is calculation correct?

Comment (faboweb) [reply src/components/common/ActionModal.vue]: Looks good!

File: test/unit/specs/components/common/ActionModal.spec.js [hunk 2]
```diff
@@ -271,6 +271,27 @@ describe(`ActionModal`, () => {
     })
   })
 
+  describe(`validates total price deos not exceed available atoms`, () => {
```
Comment (jbibla) [inline test/unit/specs/components/common/ActionModal.spec.js]:
```suggestion
  describe(`validates total price does not exceed available atoms`, () => {
```

**Comment by jbibla** [comment]
looks good. just a little typo.

**Comment by colw** [comment]
> looks good. just a little typo.

Woops. Thanks!

**Pull request merged**